src/components/viewers/MultiFlatmap.vue

```diff
@@ -145,21 +149,92 @@ export default {
       }
     },
     multiFlatmapReady: function (component) {
-      if (this.syncMode == true) 
-        this.$refs.multiflatmap.getCurrentFlatmap().enablePanZoomEvents(true);
+      this.$refs.multiflatmap.getCurrentFlatmap().enablePanZoomEvents(true); // Use zoom events for dynamic markers
       this.updateMarkers(component);
       //component.addPanZoomEvent();
     },
+    updateMarkers: function(component) {
+      let map = component.mapImp
+      map.clearMarkers();
+      let params = [];
+      if (this.apiLocation) {
+        store.state.settings.facets.species.forEach(e => {
+          params.push(encodeURIComponent('species') + '=' + encodeURIComponent(e));
+        });
+        if (this._controller) 
+          this._controller.abort();
+        this._controller = new AbortController();
+        let signal = this._controller.signal;
+        fetch(`${this.apiLocation}get-organ-curies?${params.join('&')}`, {signal})
+        .then((response) => response.json())
+        .then((data) => {
+          this.idNamePair = {}
+          this._controller = undefined;
+          data.uberon.array.forEach((pair) => {
+            let id = pair.id.toUpperCase();
+            this.idNamePair[id] = pair.name.charAt(0).toUpperCase() + pair.name.slice(1);
+          });
+          this.markers = {...this.idNamePair}
+          this.flatmapMarkerZoomUpdate()
+        })
+        .catch(err=> {
+          if (err.name !== 'AbortError') {
+            let terms = getAvailableTermsForSpecies(map.describes);
+            for (let i = 0; i < terms.length; i++) {
+              map.addMarker(terms[i].id, terms[i].type);
+            }
+          }
+        });
+      } else {
+        let terms = getAvailableTermsForSpecies(map.describes);
+        for (let i = 0; i < terms.length; i++) {
+          map.addMarker(terms[i].id, terms[i].type);
+        }
+      }
+    },
+    flatmapMarkerZoomUpdate(){
+      let flatmapImp = this.getFlatmapImp()
```

Comment:
Calling the clearMarkers and addMarker at every zoom event can be intensive. Can we store the current zoom level and only update the markers when the zoom level has changed?

Reply:
yeah good idea